plugins/clipboard/plugin.js

```diff
@@ -2537,8 +2590,395 @@
+
+		/**
+		 * A MIME type used for storing custom MIME types.
+		 *
+		 * @private
+		 * @property {String} [_customDataFallbackType='text/html']
+		 */
+		this._customDataFallbackType = 'text/html';
+	};
+
+	/**
+	 * True if the environment supports custom MIME types in {@link CKEDITOR.plugins.clipboard.dataTransfer#getData}
+	 * and {@link CKEDITOR.plugins.clipboard.dataTransfer#setData} methods.
+	 *
+	 * Introduced to distinguish browsers which supports only some whitelisted types (like `text/html`, `application/xml`),
+	 * but does not support custom MIME types (like `cke/id`). When the value of this property equals `null`
+	 * it means it was not yet initialized.
+	 *
+	 * This property should not be accessed directly, use {@link #isRequired} method instead.
+	 *
+	 * @private
+	 * @static
+	 * @property {Boolean}
+	 */
+	CKEDITOR.plugins.clipboard.fallbackDataTransfer._isCustomMimeTypeSupported = null;
+
+	CKEDITOR.plugins.clipboard.fallbackDataTransfer.prototype = {
+		/**
+		 * Whether {@link CKEDITOR.plugins.clipboard.fallbackDataTransfer fallbackDataTransfer object} should
+		 * be used when operating on native `dataTransfer`. If `true` is returned, it means custom MIME types
+		 * are not supported in the current browser (see {@link #_isCustomMimeTypeSupported}).
+		 *
+		 * @returns {Boolean}
+		 */
+		isRequired: function() {
+			var fallbackDataTransfer = CKEDITOR.plugins.clipboard.fallbackDataTransfer;
+
+			if ( fallbackDataTransfer._isCustomMimeTypeSupported === null ) {
+				// If there is no `dataTransfer` we cannot detect if fallback is needed.
+				// Method returns `false` so regular flow will be applied.
+				if ( !this._nativeDataTransfer ) {
+					return false;
+				} else {
+					var testValue = 'cke test value',
+						testType = 'cke/mimetypetest';
+
+					fallbackDataTransfer._isCustomMimeTypeSupported = false;
+
+					try {
+						this._nativeDataTransfer.setData( testType, testValue );
+						fallbackDataTransfer._isCustomMimeTypeSupported = this._nativeDataTransfer.getData( testType ) === testValue;
+						this._nativeDataTransfer.clearData( testType );
+					} catch ( e ) {}
+				}
+			}
+			return !fallbackDataTransfer._isCustomMimeTypeSupported;
+		},
+
+		/**
+		 * Returns the data of the given MIME type if stored in a regular way or in a special comment. If given type
+		 * is the same as {@link #_customDataFallbackType} the whole data without special comment is returned.
+		 *
+		 * @param {String} type
+		 * @returns {String}
+		 */
+		getData: function( type ) {
+			// As cache is already checked in CKEDITOR.plugins.clipboard.dataTransfer#getData it is skipped
+			// here. So the assumption is the given type is not in cache.
+
+			var dataComment = this._extractDataComment( this._getData( this._customDataFallbackType, true ) ),
+				value = null;
+
+			// If we are getting the same type which may store custom data we need to extract content only.
+			if ( type === this._customDataFallbackType ) {
```

Comment:
What about plain text? AFAICT in this construction it will prioritize looking for plain text in the `dataComment`.

Reply:
Yes, it will. But as we don't have hardcoded list of supported types, first checking `dataComment` is required. Remember that accessing unsupported type in Edge returns empty string (no exception so it can't be distinguished properly). As `plain/text` won't be there it will simply fallback to regular `getData`.

Reply:
Make sense 👍